### ISSUE TITLE
Pi-stomp on PI5 would be awesome!!!

### ISSUE THREAD

**Issue by maarthome** [TreeFallSound/pi-stomp#97]
Hi all,

Pi-stomp on PI5 would be awesome!!!
Best wishes to you all!!

Maarten

**Comment by rreichenbach** [comment]
Workin' on it.  The new RP1 GPIO management screws up a few things (SPI and PWM specifically).  No more RPi.GPIO.  I've been converting pi-Stomp code to use gpiozero instead which is probably better in the long run.  Still some low level things that will likely require kernel changes though.

**Comment by maarthome** [comment]
Hi Randall,

Is there any update?
I am sooooo anxious to try Pi-Stomp on my Pi5! :-)

Thanks a lot for working on this great project!

**Comment by JustBKen** [comment]
Still stuck with RP1 PWM support for the LEDs. Devs are waiting on this: https://github.com/jgarff/rpi_ws281x/issues/528

**Comment by JustBKen** [comment]
Everything else works with a PI 5 tho. I am running pi5 on my test build, just without the LEDs.

**Comment by maarthome** [comment]
That is great! Looking forward tp try it....
Enjoy 😃🙏

**Comment by maarthome** [comment]
How did you get it installed?
I have hardware version 2....
Could you give me your procedure?

Thanks!
Maarten

**Comment by rreichenbach** [comment]
If you mean, how to get bookworm/pi5 to work with v2 hardware, you can do this for now:

Start with bookworm 64-bit lite
```
sudo apt install -y git
git clone -b beta-bookworm-dev https://github.com/treefallsound/pi-stomp.git](https://github.com/micahvdm/pi-stomp.git
cd pi-stomp
nohup ./setup.sh -v 2 > setup.log | tail -f setup.log
```

After reboot
```
sudo apt remove python3-rpi.gpio
sudo apt install python3-rpi-lgpio
```

Add this line to /boot/firmware/config.txt:
`dtoverlay=spi0-2cs,cs0_pin=0,cs1_pin=7`

`sudo reboot`

**Comment by JustBKen** [comment]
If that fails, do not remove rpi.gpio

**Comment by maarthome** [comment]
Hi Randall,

Thanks for your fast reply!!!
The whole compilation goes OK now... 😊😊
However, the screen is switched on/off displaying 'pi Stomp' in green or nothing on a glitch screen...

Perhaps you have some time to look at the errors... ? 🎶
[errorlog.txt](https://github.com/user-attachments/files/16709681/errorlog.txt)

PS I checked whether the pre-bookworm version still worked... and indeed... it did (although the display 'stuttered' before the UI became visible)

Thanks again and cheers,

Maarten

**Comment by JustBKen** [comment]
Could you post the outputs of 

ps-run -l debug
sudo journalctl -fu mod-ui
sudo journalctl -fu mod-host 
sudo journalctl -fu jack

**Comment by rreichenbach** [comment]
Looks like it's having a problem controlling the GPIO's for the LED pins.  Not sure if it's a problem with the LED module for gpiozero or the way we're using it, but I only hit that issue using a pi5 not a pi4 so I'm wondering if there's a RP1 issue with gpiozero.

Anyway, for a workaround, you should probably be able to comment out or delete the gpio_output lines in ~/data/config/default_config.yml, then ps-restart.   Your LED halos (if you have any) won't light, but on my build, everything else seems to work.

**Comment by maarthome** [comment]
Hi Randall,
Thanks for your quick and VERY helpful replies!
It works now... sooooo glad!

Also 'Birge "Ken" Tok' ... Thanks for monitoring and helping!!

Next is.... making it possible to select models and IR's via the encoder and display...

Thanks again,
Greetings,
Maarten

**Comment by rreichenbach** [comment]
Yeah, good point about model/IR selection, Maarten.  And yes, thanks Ken for your help!

Those parameters are different than what we currently support for editing via the LCD.  They are of LV2 type atom:Path.  Might be a little more complex, but if MOD can grok the info for the MOD-UI, we should be able too.

**Comment by maarthome** [comment]
Hi Randall,

Strange things is that with the ToobAmpML very good lv2 plugin set from #https://github.com/rerdavies/ToobAmp (Robin Davies)  in PISTOMP, it is possible to load models via the LCD screen but NOT via the web interface... ? (at least I can load models here..... 😊) (You should checkout these plugins.... also the flanger is absolutely perfect!)

Cheers,
Maarten

**Comment by rreichenbach** [comment]
Wow, cool plugin.  Thanks for sharing that.  Will need to add that to our defaults.

The models that I see available in the TooB ML Amplifier plugin (BluesJr, BossDistortionDS1, etc.) are "hardcoded" as an lv2:enumeration

https://github.com/rerdavies/ToobAmp/blob/a9a5b60dbfd5234c1a0081d894b2c3b5de8ba885/src/ToobAmp.lv2/ttl.in/ToobML.ttl.in#L171

but Not a dynamic list based on which files are loaded as far as I can tell.  That selection does work via LCD or MOD-UI for me, cuz yeah, we do support enumerations.

**Comment by hats-np** [comment]
I apologize for the necro, but I was wondering about the Pi5 and also about the Radxa X4 as an alternative. It's a bit more expensive but it has much more processing power, x86 so there's more choices for distro, a built in RP2040, and is about the same form factor. would it even be doable?

**Comment by rreichenbach** [comment]
pi-Stomp v3 was designed around and ships with a pi5
